From: Michael Froman <mfroman@mozilla.com>
Date: Mon, 19 Jan 2026 13:43:18 -0600
Subject: (tmp-cherry-pick) Revert "Update PeerConnectionInterfaceTest and
 FakePortAllocator" (42bfa2a946)

This reverts commit 7a56759ba5e8d5f8c8a84c3417b12304ebafbf67.

Reason for revert: Broke Chromium link step.

See https://ci.chromium.org/ui/p/chromium/builders/try/linux_chromium_compile_dbg_ng/2630200/overview

Original change's description:
> Update PeerConnectionInterfaceTest and FakePortAllocator
>
> Use WeakPtr instead of a potentially dangling pointer to the
> PortAllocator. Once Close() has been called, it is possible that the
> port allocator has been deleted (although in the current implementation
> it is not).
>
> Add callback when candidate pool is discarded rather than dereferene the
> fake_port_allocator_ pointer.
>
> Bug: None
> Change-Id: I89ee155e62f13f8539f9c8a67bd87c977ae65d16
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/427362
> Reviewed-by: Harald Alvestrand <hta@webrtc.org>
> Commit-Queue: Tomas Gunnarsson <tommi@webrtc.org>
> Auto-Submit: Tomas Gunnarsson <tommi@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46260}

Bug: None
No-Presubmit: true
No-Try: true
No-Tree-Checks: true
Change-Id: I6303c3bb9ab1f2fa8ca0e271ce1225e40b6c817b
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/427660
Auto-Submit: Harald Alvestrand <hta@webrtc.org>
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#46266}
---
 p2p/BUILD.gn                             |  2 --
 p2p/base/port_allocator.h                |  2 +-
 p2p/test/fake_port_allocator.h           | 20 ------------------
 pc/BUILD.gn                              |  1 -
 pc/peer_connection_interface_unittest.cc | 27 ++++++++----------------
 5 files changed, 10 insertions(+), 42 deletions(-)

diff --git a/p2p/BUILD.gn b/p2p/BUILD.gn
index c435a2fc78..91e7ba7361 100644
--- a/p2p/BUILD.gn
+++ b/p2p/BUILD.gn
@@ -1075,9 +1075,7 @@ if (rtc_include_tests) {
       "../rtc_base:network",
       "../rtc_base:socket_factory",
       "../rtc_base:task_queue_for_test",
-      "../rtc_base:weak_ptr",
       "//third_party/abseil-cpp/absl/base:nullability",
-      "//third_party/abseil-cpp/absl/functional:any_invocable",
       "//third_party/abseil-cpp/absl/memory",
       "//third_party/abseil-cpp/absl/strings:string_view",
     ]
diff --git a/p2p/base/port_allocator.h b/p2p/base/port_allocator.h
index be91dae6d8..1b8661ed92 100644
--- a/p2p/base/port_allocator.h
+++ b/p2p/base/port_allocator.h
@@ -510,7 +510,7 @@ class RTC_EXPORT PortAllocator {
       const IceParameters* ice_credentials = nullptr) const;
 
   // Discard any remaining pooled sessions.
-  virtual void DiscardCandidatePool();
+  void DiscardCandidatePool();
 
   // Clears the address and the related address fields of a local candidate to
   // avoid IP leakage. This is applicable in several scenarios:
diff --git a/p2p/test/fake_port_allocator.h b/p2p/test/fake_port_allocator.h
index 26bf9d06dc..c1ce3518ed 100644
--- a/p2p/test/fake_port_allocator.h
+++ b/p2p/test/fake_port_allocator.h
@@ -13,11 +13,9 @@
 
 #include <cstdint>
 #include <memory>
-#include <utility>
 #include <vector>
 
 #include "absl/base/nullability.h"
-#include "absl/functional/any_invocable.h"
 #include "absl/memory/memory.h"
 #include "absl/strings/string_view.h"
 #include "api/candidate.h"
@@ -36,7 +34,6 @@
 #include "rtc_base/network.h"
 #include "rtc_base/socket_factory.h"
 #include "rtc_base/task_queue_for_test.h"
-#include "rtc_base/weak_ptr.h"
 
 namespace webrtc {
 
@@ -256,32 +253,15 @@ class FakePortAllocator : public PortAllocator {
   bool MdnsObfuscationEnabled() const override {
     return mdns_obfuscation_enabled_;
   }
-
   void SetMdnsObfuscationEnabledForTesting(bool enabled) {
     mdns_obfuscation_enabled_ = enabled;
   }
 
-  void DiscardCandidatePool() override {
-    PortAllocator::DiscardCandidatePool();
-    if (on_candidate_pool_discarded_) {
-      std::move(on_candidate_pool_discarded_)();
-    }
-  }
-
-  void SetOnDiscardCandidatePool(
-      absl::AnyInvocable<void() &&> on_candidate_pool_discarded) {
-    on_candidate_pool_discarded_ = std::move(on_candidate_pool_discarded);
-  }
-
-  WeakPtr<FakePortAllocator> NewWeakPtr() { return weak_factory_.GetWeakPtr(); }
-
  private:
   const Environment env_;
   TaskQueueBase* absl_nonnull network_thread_;
   BasicPacketSocketFactory factory_;
   bool mdns_obfuscation_enabled_ = false;
-  absl::AnyInvocable<void() &&> on_candidate_pool_discarded_;
-  WeakPtrFactory<FakePortAllocator> weak_factory_{this};
 };
 
 }  //  namespace webrtc
diff --git a/pc/BUILD.gn b/pc/BUILD.gn
index 522ed06057..3a8246e8f9 100644
--- a/pc/BUILD.gn
+++ b/pc/BUILD.gn
@@ -2701,7 +2701,6 @@ if (rtc_include_tests && !build_with_chromium) {
       "../rtc_base:threading",
       "../rtc_base:timeutils",
       "../rtc_base:unique_id_generator",
-      "../rtc_base:weak_ptr",
       "../rtc_base/containers:flat_map",
       "../rtc_base/synchronization:mutex",
       "../system_wrappers",
diff --git a/pc/peer_connection_interface_unittest.cc b/pc/peer_connection_interface_unittest.cc
index ddd7cbe705..9d41ad2af0 100644
--- a/pc/peer_connection_interface_unittest.cc
+++ b/pc/peer_connection_interface_unittest.cc
@@ -42,7 +42,6 @@
 #include "api/rtp_sender_interface.h"
 #include "api/rtp_transceiver_direction.h"
 #include "api/scoped_refptr.h"
-#include "api/sctp_transport_interface.h"
 #include "api/test/rtc_error_matchers.h"
 #include "api/transport/bitrate_settings.h"
 #include "api/transport/enums.h"
@@ -60,6 +59,7 @@
 #include "media/base/codec.h"
 #include "media/base/media_config.h"
 #include "media/base/stream_params.h"
+#include "media/sctp/sctp_transport_internal.h"
 #include "p2p/base/p2p_constants.h"
 #include "p2p/base/port.h"
 #include "p2p/base/port_allocator.h"
@@ -88,7 +88,6 @@
 #include "rtc_base/socket_server.h"
 #include "rtc_base/thread.h"
 #include "rtc_base/virtual_socket_server.h"
-#include "rtc_base/weak_ptr.h"
 #include "test/gmock.h"
 #include "test/gtest.h"
 #include "test/wait_until.h"
@@ -102,7 +101,6 @@ namespace {
 
 using ::testing::Eq;
 using ::testing::Exactly;
-using ::testing::IsNull;
 using ::testing::IsTrue;
 using ::testing::NotNull;
 using ::testing::SizeIs;
@@ -662,12 +660,8 @@ class PeerConnectionInterfaceBaseTest : public ::testing::Test {
   }
 
   void TearDown() override {
-    if (pc_) {
+    if (pc_)
       pc_->Close();
-      ReleasePeerConnection();
-    }
-
-    ASSERT_THAT(port_allocator_, IsNull());
   }
 
   void CreatePeerConnection() {
@@ -708,12 +702,11 @@ class PeerConnectionInterfaceBaseTest : public ::testing::Test {
   void CreatePeerConnection(const RTCConfiguration& config) {
     if (pc_) {
       pc_->Close();
-      ReleasePeerConnection();
+      pc_ = nullptr;
     }
-    ASSERT_THAT(port_allocator_, IsNull());
     auto port_allocator =
         std::make_unique<FakePortAllocator>(CreateEnvironment(), vss_.get());
-    port_allocator_ = port_allocator->NewWeakPtr();
+    port_allocator_ = port_allocator.get();
 
     // Create certificate generator unless DTLS constraint is explicitly set to
     // false.
@@ -1259,7 +1252,7 @@ class PeerConnectionInterfaceBaseTest : public ::testing::Test {
   std::unique_ptr<VirtualSocketServer> vss_;
   AutoSocketServerThread main_;
   scoped_refptr<FakeAudioCaptureModule> fake_audio_capture_module_;
-  WeakPtr<FakePortAllocator> port_allocator_;
+  FakePortAllocator* port_allocator_ = nullptr;
   FakeRTCCertificateGenerator* fake_certificate_generator_ = nullptr;
   scoped_refptr<PeerConnectionFactoryInterface> pc_factory_;
   scoped_refptr<PeerConnectionInterface> pc_;
@@ -1361,7 +1354,7 @@ TEST_P(PeerConnectionInterfaceTest,
   // Create fake port allocator.
   auto port_allocator =
       std::make_unique<FakePortAllocator>(CreateEnvironment(), socket_server());
-  WeakPtr<FakePortAllocator> raw_port_allocator = port_allocator->NewWeakPtr();
+  FakePortAllocator* raw_port_allocator = port_allocator.get();
 
   // Create RTCConfiguration with some network-related fields relevant to
   // PortAllocator populated.
@@ -2327,16 +2320,14 @@ TEST_P(PeerConnectionInterfaceTest,
 TEST_P(PeerConnectionInterfaceTest, PooledSessionsDiscardedAfterClose) {
   CreatePeerConnection();
 
-  bool candidate_pool_discarded = false;
-  port_allocator_->SetOnDiscardCandidatePool(
-      [&]() { candidate_pool_discarded = true; });
-
   PeerConnectionInterface::RTCConfiguration config = pc_->GetConfiguration();
   config.ice_candidate_pool_size = 3;
   EXPECT_TRUE(pc_->SetConfiguration(config).ok());
   pc_->Close();
 
-  EXPECT_TRUE(candidate_pool_discarded);
+  // Expect no pooled sessions to be left.
+  const PortAllocatorSession* session = port_allocator_->GetPooledSession();
+  EXPECT_EQ(nullptr, session);
 }
 
 // Test that SetConfiguration returns an invalid modification error if
